### PR TITLE
Add multiple optimizers to train config

### PR DESCRIPTION
**Pull request by robert-wagner**
There also apparently are some connivence changes that I forgot to merge back. Whoops

### CODE REVIEW

File: examples/tutorials/advanced/Federated Learning with TrainConfig/Introduction to TrainConfig.ipynb [hunk 1]
```diff
@@ -297,7 +298,7 @@
    "outputs": [],
    "source": [
     "batch_size = 4\n",
-    "lr = 0.1\n",
+    "optimizer_args = {\"lr\" : 0.1, \"weight_decay\" : 0.01}\n",
```
Comment (mari-linhares) [inline examples/tutorials/advanced/Federated Learning with TrainConfig/Introduction to TrainConfig.ipynb]: nice, I really like this :)

File: syft/federated/federated_client.py [hunk 2]
```diff
@@ -34,27 +34,22 @@ def set_obj(self, obj: object):
...
-        self, optimizer_name: str, model, lr: float, weight_decay: float
+        self, optimizer_name: str, model, optimizer_args: dict
     ) -> th.optim.Optimizer:
         """Build an optimizer if needed.
 
         Args:
             optimizer_name: A string indicating the optimizer name.
-            lr: A float indicating the learning rate.
-            weight_decay: Weight decay parameter of the optimizer
+            optimizer_args: A dict containing the args used to initialize the optimizer
```
Comment (mari-linhares) [inline syft/federated/federated_client.py]: Missing period.

File: syft/federated/train_config.py [hunk 3]
```diff
@@ -41,7 +40,7 @@ def __init__(
             batch_size: Batch size used for training.
             epochs: Epochs used for training.
             optimizer: A string indicating which optimizer should be used.
-            lr: Learning rate.
+            optimizer_args: A dict containing the arguments to initialize the optimizer
```
Comment (mari-linhares) [inline syft/federated/train_config.py]: Missing period. Add a sentence about default values.

File: examples/tutorials/advanced/Federated Learning with TrainConfig/Introduction to TrainConfig.ipynb [hunk 4]
```diff
@@ -282,9 +283,9 @@
    "source": [
     "### General hyper parameters and training options\n",
     "\n",
-    "Se currently support:\n",
+    "TrainCongig currently support:\n",
```
Comment (midokura-silvia) [inline examples/tutorials/advanced/Federated Learning with TrainConfig/Introduction to TrainConfig.ipynb]: typo

File: syft/federated/train_config.py [hunk 5]
```diff
@@ -23,14 +23,13 @@ def __init__(
         owner: workers.AbstractWorker = None,
         batch_size: int = 32,
         epochs: int = 1,
-        optimizer: str = "sgd",
-        lr: float = 0.1,
+        optimizer: str = "SGD",
+        optimizer_args: dict = {"lr": 0.1, "weight_decay": 0.01},
```
Comment (midokura-silvia) [inline syft/federated/train_config.py]: I would tend to setting the weight_decay by default to None.

Comment (robert-wagner) [reply syft/federated/train_config.py]: The weight decay was originally defaulted to 0.01 (before this pr). I am not opposed to setting it to None (in this case just not included) but I wanted to maintain the original behavior (with that in mind, what is your preference)

Comment (mari-linhares) [reply syft/federated/train_config.py]: I vote for None.

File: syft/federated/train_config.py [hunk 6]
```diff
@@ -41,7 +40,7 @@ def __init__(
             batch_size: Batch size used for training.
             epochs: Epochs used for training.
             optimizer: A string indicating which optimizer should be used.
-            lr: Learning rate.
+            optimizer_args: A dict containing the arguments to initialize the optimizer. Defaults fot {'lr': 0.1, 'weight_decay': 0.01}.
```
Comment (mari-linhares) [inline syft/federated/train_config.py]: fot?

Comment (robert-wagner) [reply syft/federated/train_config.py]: whoops

File: syft/federated/train_config.py (hunk 5 above)
Comment (midokura-silvia) [reply syft/federated/train_config.py]: I added the weight decay and I also vote for None (default value of the sgd optimizer). That's how I thought I left it.
